proposals/20230605-rules-adoption-management-maturity-framework.md

```diff
@@ -0,0 +1,222 @@
+
+
+The minimum criteria for rules creation shall include:
+
+*Correctness*
+
+The rule must be both syntactically and grammatically correct and should evaluate to true during successful end-to-end tests. Furthermore, it needs to accurately detect the intended cyber threats, specifically the Tactics, Techniques, and Procedures (TTPs).
+
+
+*Robustness*
+
+To enhance the effectiveness of detection, priority is given to behavioral detections, as opposed to simple string matching on process command arguments or other fields. This preference is based on the ease with which the latter can be circumvented. The same principle applies when selecting the most robust system call for detecting a specific threat at a particular point in time or attack scenario. For concrete examples of more robust rules, please refer to Appendix 1.
+
+
+*Relevance*
+
+Determining relevance is often the most subjective criterion, as it requires expertise in offensive security, cyber defense, and real-world production settings for accurate assessment. Questions such as whether these threats are a priority for most organizations or if we can provide enough context for a security analyst to appropriately act on the alerts as part of their incident response workflows are top of mind when assessing the overall relevance. Relevance is a key factor that indirectly reflects both robustness and significance, but more importantly, it indicates whether a particular security threat is significant to most adopters and, consequently, beneficial to detect.
+
+Possible criteria:
+
+- Cover relevant attack vectors across various industries.
+- Emphasize profiling over signatures.
+- Effectiveness of the rule across diverse workloads.
+- Guidance and templates are provided to assist with tuning.
+
+*Testing*
+
+How was the rule tested? The desired testing approach includes not only functional end-to-end tests in virtual machines (VMs) but also deployments in real-world infrastructure. The Falco maintainers will provide guidance and support to contributors throughout the testing process, recognizing that it will vary significantly for each rule due to its unique nature.
+
+
+### Guidance on Rule Customization and Tuning
+
+Each rule tagged as "Stable" will provide clear guidance on how it can be tuned, customized, or combined with other rules if applicable. Over time, a catalog of general best tuning practices will be developed. For specific examples, please refer to Appendix 2. 
+
+
+### Setting Expectations for Adopters
+
+Setting clear expectations for adopters of Falco is crucial. While Falco is a powerful security monitoring tool, it's important to be aware of its limitations and maintain realistic expectations. Please refer to Appendix 3 for specific examples of current limitations of Falco.
```

Comment:
What do you think if we also guide the community to lateral tools for specific use cases Falco does not cover?
For example, in case the community user would need projects for threat detection with network analysis, could be guided to projects like Suricata.

Reply:
This would be great! If I recall correctly, we wanted to add some more generic educational resources on the website as well, also around eBPF or any topic / domain falco touches.